shared-bindings/analogio/AnalogFastIn.c

```diff
@@ -0,0 +1,167 @@
+    common_hal_analogio_analogfastin_construct(self,
+        pin,
+        ((uint8_t *)bufinfo.buf),
+        bufinfo.len,
+        bytes_per_sample,
+        signed_samples,
+        sample_rate
+        );
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+//|     def deinit(self) -> None:
+//|         """Turn off the AnalogFastIn and release the pin for other use."""
+//|         ...
+//|
+STATIC mp_obj_t analogio_analogfastin_deinit(mp_obj_t self_in) {
+    analogio_analogfastin_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    common_hal_analogio_analogfastin_deinit(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(analogio_analogfastin_deinit_obj, analogio_analogfastin_deinit);
+
+STATIC void check_for_deinit(analogio_analogfastin_obj_t *self) {
+    if (common_hal_analogio_analogfastin_deinited(self)) {
+        raise_deinited_error();
+    }
+}
+//|     def __enter__(self) -> AnalogFastIn:
+//|         """No-op used by Context Managers."""
+//|         ...
+//|
+//  Provided by context manager helper.
+
+//|     def __exit__(self) -> None:
+//|         """Automatically deinitializes the hardware when exiting a context. See
+//|         :ref:`lifetime-and-contextmanagers` for more info."""
+//|         ...
+//|
+STATIC mp_obj_t analogio_analogfastin___exit__(size_t n_args, const mp_obj_t *args) {
+    (void)n_args;
+    common_hal_analogio_analogfastin_deinit(args[0]);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(analogio_analogfastin___exit___obj, 4, 4, analogio_analogfastin___exit__);
+
+//|     value: int
+//|     """The value on the analog pin between 0 and 65535 inclusive (16-bit). (read-only)
+//|
+//|     Even if the underlying analog to digital converter (ADC) is lower
+//|     resolution, the value is 16-bit."""
+//|
+STATIC mp_obj_t analogio_analogfastin_obj_capture(mp_obj_t self_in) {
+    analogio_analogfastin_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+    return MP_OBJ_NEW_SMALL_INT(common_hal_analogio_analogfastin_capture(self));
+}
+MP_DEFINE_CONST_FUN_OBJ_1(analogio_analogfastin_capture_obj, analogio_analogfastin_obj_capture);
+
+// MP_PROPERTY_GETTER(analogio_analogfastin_value_obj,
+//    (mp_obj_t)&analogio_analogfastin_get_value_obj);
+
+//|     reference_voltage: float
+//|     """The maximum voltage measurable (also known as the reference voltage) as a
+//|     `float` in Volts.  Note the ADC value may not scale to the actual voltage linearly
+//|     at ends of the analog range."""
+//|
+
+STATIC const mp_rom_map_elem_t analogio_analogfastin_locals_dict_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_deinit),             MP_ROM_PTR(&analogio_analogfastin_deinit_obj) },
+    { MP_ROM_QSTR(MP_QSTR___enter__),          MP_ROM_PTR(&default___enter___obj) },
+    { MP_ROM_QSTR(MP_QSTR___exit__),           MP_ROM_PTR(&analogio_analogfastin___exit___obj) },
+    { MP_ROM_QSTR(MP_QSTR_capture),            MP_ROM_PTR(&analogio_analogfastin_capture_obj)},
```

Comment:
A question, but would the user need to get the reference voltage like you find in AnalogIn?